### PR TITLE
Bug-2054841: Increase services unit test coverage to 100%

### PR DESCRIPTION
**Pull request by moijes12**
This PR adds comprehensive unit test coverage for the `treeherder/services/`  package to achieve 100% line coverage:

## Changes

**Test Files:**
- `tests/services/pulse/test_consumers.py`: Added tests covering all consumer classes (TaskConsumer, MozciClassificationConsumer, PushConsumer, JointConsumer), factory functions (prepare_consumers, prepare_joint_consumers), and edge cases like binding/unbinding queues and stale binding pruning
- `tests/services/pulse/test_exchange.py`: Added test for non-existent exchange error handling with create=False
- `tests/services/test_taskcluster.py`: Added tests covering TaskclusterModelImpl.trigger_action(), _load(), _submit(), hook submission, scope validation, and factory methods

**Configuration:**
- `tests/settings.py`: Added memory-based cache backend (LocMemCache) and in-memory Celery broker URL (memory://) to enable isolated test execution without external dependencies (Redis, RabbitMQ)

## Coverage
Achieves 100% line coverage for:
- `treeherder/services/pulse/consumers.py`
- `treeherder/services/pulse/exchange.py`
- `treeherder/services/taskcluster.py`

image

Fixes [Bug-2054841](https://bugzilla.mozilla.org/show_bug.cgi?id=2054841)

@gmierz @Archaeopteryx

### CODE REVIEW

File: tests/services/test_taskcluster.py [hunk 1]
```diff
@@ -15,21 +18,25 @@
...
 
 @pytest.fixture(scope="module")
 def original_task():
+    """Load original task sample fixture data."""
     return load_json_fixture("originalTask.json")
 
 
 @pytest.fixture(scope="module")
 def expected_backfill_task():
+    """Load expected backfill task sample fixture data."""
```
Comment (gmierz) [inline tests/services/test_taskcluster.py]: These comments don't seem necessary to me since the code is already pretty descriptive about what they do.

File: tests/services/pulse/test_consumers.py [hunk 2]
```diff
@@ -125,3 +147,342 @@ def mock_store_pulse_tasks_classification(args, queue):
     consumer.on_message(None, message)
 
     assert mock_called
+
+
+# Additional Unit Tests for 100% Coverage of treeherder/services/pulse/consumers.py
```
Comment (gmierz) [inline tests/services/pulse/test_consumers.py]: Remove this please.

**Review by gmierz — CHANGES_REQUESTED** [review]
Looks good, thanks! Just need to rework the comments a bit.

File: tests/services/test_taskcluster.py (hunk 1 above)
Comment (moijes12) [reply tests/services/test_taskcluster.py]: Addressed in latest commit

File: tests/services/pulse/test_consumers.py (hunk 2 above)
Comment (moijes12) [reply tests/services/pulse/test_consumers.py]: Addressed in latest commit

**Comment by moijes12** [comment]
@gmierz The changes requested have been addressed in the latest commit. The PR is ready for review again.

**Review by gmierz — APPROVED** [review]
r+ thanks for the new tests!
